test/protocol/disconnect_dap_test.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require_relative '../support/protocol_test_case'
+
+module DEBUGGER__
+
+  class DisconnectDAPTest < ProtocolTestCase
+    PROGRAM = <<~RUBY
+       1| module Foo
+       2|   class Bar
+       3|     def self.a
+       4|       "hello"
+       5|     end
+       6|   end
+       7|   loop do
+       8|     b = 1
+       9|   end
+      10|   Bar.a
+      11|   bar = Bar.new
+      12| end
+    RUBY
+
+    def test_disconnect_without_terminateDebuggee_keeps_debuggee_alive
+      run_protocol_scenario PROGRAM, cdp: false do
+        req_dap_disconnect(terminate_debuggee: false)
+        attach_to_dap_server
+        assert_reattached
+        # suspends the debuggee so it'll take the later requests (include terminate)
+        suspend_debugee
+        req_terminate_debuggee
+      end
+    end
+
+    def test_disconnect_with_terminateDebuggee_kills_debuggee
+      run_protocol_scenario PROGRAM, cdp: false do
+        req_dap_disconnect(terminate_debuggee: true)
+      end
+    end
+
+    private
+
+    def req_dap_disconnect(terminate_debuggee:)
```

Comment:
How about this one?

* Divide `req_disconnect` to `req_disconnect_dap_server` and `req_disconnect_cdp_server`.
* Configure `terminate_debuggee: false` in `req_disconnect_dap_server` as a param